loops-arrays-objects.js

```diff
@@ -74,18 +106,37 @@ const NOVIEmployees = [
 //   { firstName: 'Mark', lastName: 'Rensen', email: 'Mark.Rensen@novi.nl' },
 //  ];
 // ==========================================
+// Stappenplan:
+// 1. loop door de array heen en voeg bij elk object :
+// a) de property email toe.
+// b) het emailadres op basis van de properties firstName en LastName.
+//
 
 
-
-
-
+for (let i= 0; i < NOVIEmployees.length; i++) {
+    NOVIEmployees[i].email = `${NOVIEmployees[i].firstName}.${NOVIEmployees[i].lastName}@novi.nl`;
+    console.log(NOVIEmployees[i]);
+}
 
 // Opdracht 2-BONUS
 // Lukt het je om ervoor te zorgen dat alle e-mailadressen in lowercase letters komen te staan? Dit heb je nog niet geleerd, maar google is your best friend...
 // ==========================================
-
-
-
+//Stappenplan:
+// 1. loop door de array heen en
+// a) zet elke string om in kleine letters
+// b) voeg bij elk object :
+//    b1) de property email toe.
+//    b2) het emailadres op basis van de properties firstName en LastName.
+
+
+for (let i = 0; i < NOVIEmployees.length; i++) {
+    let firstNameSmall= NOVIEmployees[i].firstName.toLowerCase();
+    NOVIEmployees[i].firstName = firstNameSmall;
+    let lastNameSmall = NOVIEmployees[i].lastName.toLowerCase();
+    NOVIEmployees[i].lastName = lastNameSmall;
+    NOVIEmployees[i].email = `${firstNameSmall}.${lastNameSmall}@novi.nl`;
+    console.log(NOVIEmployees[i]);
+}
 
```

Comment:
Je zet hier elke waarde van een object om in kleine letters. Alleen het emaildres was voldoende. Het kan dus iets korter:
``` 
for (let i = 0; i < NOVIEmployees.length; i++) {
    const emailaddress = NOVIEmployees[i].firstName + '.' + NOVIEmployees[i].lastName + '@novi.nl';
    NOVIEmployees[i].email = emailaddress.toLowerCase();
}
```
Ook hier is een let niet nodig, waarde binnen een herhaling van de for loop verandert niet.